packages/@aws-cdk/aws-vpcv2-alpha/lib/vpc-v2.ts

```diff
@@ -0,0 +1,432 @@
+import { CfnVPC, CfnVPCCidrBlock, ISubnet } from 'aws-cdk-lib/aws-ec2';
+import { Arn, CfnResource, Lazy, Names } from 'aws-cdk-lib/core';
+import { Construct, DependencyGroup, IDependable } from 'constructs';
+import { IpamIpv4, IpamPool, IpamIpv6 } from './ipam';
+import { VpcV2Base } from './vpc-v2-base';
+
+/**
+ * Ipam Options to add IPv4 CIDR range to the VPC
+ */
+export interface IIpIpamOptions {
+  /**
+   * The IPv4 IPAM pool to use for allocating IPv4 addresses.
+   *
+   * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#configuring-ipv4-ipam}.
+   */
+  readonly ipv4IpamPool?: IpamPool;
+
+  /**
+   * The IPv4 netmask length to use for allocating IPv4 addresses.
+   *
+   * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#configuring-ipv4-netmask-length}.
+   */
+  readonly ipv4NetmaskLength?: number;
+
+  /**
+   * The IPv6 IPAM pool to use for allocating IPv6 addresses.
+   *
+   * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#configuring-ipv6-ipam}.
+   */
+  readonly ipv6IpamPool?: IpamPool;
+
+  /**
+   * The IPv6 netmask length to use for allocating IPv6 addresses.
+   *
+   * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#configuring-ipv6-netmask-length}.
+   */
+  readonly ipv6NetmaskLength?: number;
+}
+
+/**
+ * IpAddress options to define VPC V2
+ */
+export class IpAddresses {
+
+  /**
+   * An IPv4 CIDR Range
+   */
+  public static ipv4(ipv4Cidr: string): IIpAddresses {
+    return new ipv4CidrAllocation(ipv4Cidr);
+  }
```

Comment:
because we do not yet support BYOIP ipv6, it will be added once we have that feature, currently its only amazon provide ipv6 address through IPAM or in the VPC class